### PR TITLE
feat(swap): Built out Select Token Modal

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Built out `Select Token Modal` and needed `Components`

### Components
- Reusable `Standard Modal`
- Reusable `Search Input`
- Updated Reusable `Select-Token` button to `Select-Token-or-Network` to be more reusable
- `Token-List` button
- `Search-With-Network-Selector` component
- `Select-Token` modal

### Shared Styled Components
- `VerticalDivider`

### Other
- `Mock-Networks`
- `Mock-Token-Balances`
- Updated `Mock-Tokens`
- Updated `index.ts` barrel files to use direct `exports`

Resolves <https://github.com/brave/brave-browser/issues/24273>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A
  
https://user-images.githubusercontent.com/40611140/181661321-c61abf3a-6e33-4497-8af4-ab8748991a8a.mov

### CODE REVIEW

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: app/mock-data/mock-networks.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    iconUrls: [SOLIconUrl],
+    rpcUrls: [''],
+    symbol: 'SOL',
+    symbolName: 'Solana'
+  },
+]
+
+export const mockEthereumNetwork = mockNetworks[0]
+export const mockBinanceNetwork = mockNetworks[1]
+export const mockSolanaNetwork = mockNetworks[2]
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply app/mock-data/mock-networks.ts]: Updated!

File: app/mock-data/mock-tokens.ts [hunk 2]
```diff
@@ -65,10 +76,100 @@ export const mockEthereumTokens: BlockchainToken[] = [
...
 export const mockEthereumToken = mockEthereumTokens[0]
 export const mockBasicAttentionToken = mockEthereumTokens[1]
 export const mockWrappedBinanceCoinToken = mockEthereumTokens[2]
 export const mockWrappedSolanaToken = mockEthereumTokens[3]
+export const mockUSDCToken = mockEthereumTokens[4]
+export const mockSUSHIToken = mockEthereumTokens[5]
+export const mockUSDTToken = mockEthereumTokens[6]
+export const mockMATICToken = mockEthereumTokens[7]
+export const mock0XBTCToken = mockEthereumTokens[8]
+export const mock1INCHToken = mockEthereumTokens[9]
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply app/mock-data/mock-tokens.ts]: Good point, updated!

File: app/src/assets/asset-icons/index.ts [hunk 3]
```diff
@@ -7,10 +7,22 @@ import BATIconUrl from './bat-asset-icon.svg'
 import BNBIconUrl from './bnb-asset-icon.png'
 import ETHIconUrl from './eth-asset-icon.svg'
 import SOLIconUrl from './sol-asset-icon.svg'
+import SUSHIIconUrl from './sushi-asset-icon.svg'
+import USDCIconUrl from './usdc-asset-icon.svg'
+import USDTIconUrl from './usdt-asset-icon.svg'
+import ZEROXBTCIconUrl from './0xbtc-asset-icon.svg'
+import MATICIconUrl from './matic-asset-icon.svg'
+import ONEINCHIconUrl from './1inch-asset-icon.svg'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply app/src/assets/asset-icons/index.ts]: Done!

File: app/src/components/buttons/token-list-button.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    onClick,
+    token,
+    balance,
+    disabled,
+    isConnected
+  } = props
+
+  const onSelectToken = React.useCallback(() => {
+    onClick(token)
+  }, [token])
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply app/src/components/buttons/token-list-button.tsx]: Added!

File: app/src/components/inputs/search-input.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  autoFocus?: boolean
+  placeholder?: string
+}
+
+export const SearchInput = (props: Props) => {
+  const { onChange, value, autoFocus, placeholder } = props
+
+  const onInputChange = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(event.target.value)
+  }, [])
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply app/src/components/inputs/search-input.tsx]: Added!

**Review by onyb — APPROVED** [review]
++

File: app/src/App.tsx [hunk 6]
```diff
@@ -79,78 +57,156 @@ function App () {
...
+
+  const insufficientBalance = React.useMemo((): boolean => {
+    if (fromAmount && fromTokenBalance !== undefined) {
+      return Number(fromAmount) > fromTokenBalance
+    }
+    return false
+  }, [fromTokenBalance, fromAmount])
+
+  const selectedToken = React.useMemo((): BlockchainToken | undefined => {
+    if (selectingFromOrTo === 'from') {
```
Comment (josheleonard) [inline app/src/App.tsx:140]: Shouldn't this be reversed?

Comment (Douglashdaniel) [reply app/src/App.tsx:140]: No shouldn’t be reversed since we don’t want to allow selecting the same token for `from` if it’s already selected for `to`

**Review by josheleonard — APPROVED** [review]

**Pull request merged**